### PR TITLE
client: Implement command timeouts

### LINKED ISSUE #88: client: Add I/O timeouts
RFC 5321:
>4.5.3.2.  Timeouts
>
>   An SMTP client MUST provide a timeout mechanism.  It MUST use per-
>   command timeouts rather than somehow trying to time the entire mail
>   transaction.  Timeouts SHOULD be easily reconfigurable, preferably
>   without recompiling the SMTP code.

### PR DESCRIPTION
**Pull request by foxcpp**
Closes #88.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -38,6 +39,12 @@ type Client struct {
 	didHello   bool     // whether we've said HELO/EHLO/LHLO
 	helloError error    // the error from the hello
 	rcpts      []string // recipients accumulated for the current session
+
+	// Time to wait for command responses (this includes 3xx reply to DATA).
+	CommandTimeout time.Duration
+
+	// Time to wait for responses after final dot.
+	DotTimeout time.Duration
```
Comment (emersion) [inline client.go:47]: How does this interact with future `CHUNKING` support?

Comment (foxcpp) [reply client.go:47]: I believe that could be mapped to the timeout for response to `BDAT ... LAST`. Do you think it should have a better name?

Comment (emersion) [reply client.go:47]: `SubmissionTimeout` maybe?

**Comment by kayrus** [comment]
Any updates on this PR?

**Comment by foxcpp** [comment]
> Any updates on this PR?

I have a lot of IRL things to do. Feel free to finish this PR if you're interested.

**Comment by foxcpp** [comment]
I renamed DotTimeout into SubmissionTimeout and also rebased PR onto upstream master so tests could run (there is new vet rule in Go 1.15 that failed)

**Review by emersion — APPROVED** [review]
LGTM, thanks!

**Pull request merged**